src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6344,8 +6344,8 @@ private ExpressionAst HashExpressionRule(Token atCurlyToken, bool parsingSchemaE
                 // Note -  the error handling function inspects the error message body to extra the ParserStrings property name. It uses this value as the errorid.
                 var errorMessageExpression = parsingSchemaElement ?
                     (Expression<Func<string>>)(() => ParserStrings.IncompletePropertyAssignmentBlock)
-                    : (Expression<Func<string>>)(() => ParserStrings.IncompleteHashLiteral);
-                ReportIncompleteInput(keyValuePairs.Any() ? After(keyValuePairs.Last().Item2) : After(atCurlyToken), rCurly.Extent, errorMessageExpression);
+                    : (Expression<Func<string>>)(() => ParserStrings.MissingEndCurlyBrace);
+                ReportIncompleteInput(After(atCurlyToken), rCurly.Extent, errorMessageExpression);
```

Comment:
This looks like it might degrade the quality of the error on:
```PowerShell
@{
a =
}
```
or something like that - please try variants of this (for example: with and without the closing curly, with and without the `=`).